tests/test_config_section.py

```diff
@@ -0,0 +1,546 @@
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+"""
+Test validating of configuration values.
+"""
+
+from base import TestBase
+from mock import MagicMock
+
+import tempfile
+import os
+from binascii import hexlify
+
+from virtwho.config import GlobalSection, VirtConfigSection, str_to_bool, VW_TYPES, \
+    MinimumSendInterval, DefaultInterval, ConfigSection, ValidationState
+from virtwho.password import Password
+from virtwho.log import DEFAULT_LOG_DIR
+
+
+# Values used for testing VirtConfigSection
+LIBVIRT_SECTION_VALUES = {
+    'type': 'libvirt',
+    'server': '10.0.0.101',
+    'username': 'admin',
+    'password': 'top_secret',
+    'env': '123456',
+    'owner': '123456',
+    'hypervisor_id': 'uuid',
+    'filter_hosts': '*.example.com'
+}
+
+# Values used for testing GlobalConfigSection
+GLOBAL_SECTION_VALUES = {
+    'debug': True,
+    'oneshot': 'true',
+    'print': 'false',
+    'wrong_bool_value': 'nein',
+    'configs': ['local_libvirt.conf'],
+    'log_file': 'my_custom.log',
+    'interval': '120'
+}
+
+
+class TestVirtConfigSection(TestBase):
+    """
+    Test base for testing class VirtConfigSection
+    """
+
+    def setUp(self):
+        """
+        Method executed before each unit test
+        """
+        self.virt_config = VirtConfigSection('test-libvirt', None)
+        # We need to set values using this way, because we need
+        # to trigger __setitem__ of virt_config
+        for key, value in LIBVIRT_SECTION_VALUES.items():
+            self.virt_config[key] = value
+
+    def test_validate_virt_type(self):
+        """
+        Test validation of supported types of virtualization backends
+        """
+        test_virt_types = list(VW_TYPES[:])
+        test_virt_types.extend('vmware,' 'kvm')
+        for virt_type in test_virt_types:
+            self.virt_config['type'] = virt_type
+            result = self.virt_config._validate_virt_type()
+            if virt_type not in VW_TYPES:
+                self.assertIsNotNone(result)
+            else:
+                self.assertIsNone(result)
+                value = self.virt_config.get('type')
+                self.assertEqual(value, virt_type)
+
+    def test_validate_missing_virt_type(self):
+        """
+        Test validation of missing type of virtualization backend
+        """
+        del self.virt_config['type']
```

Comment:
You are right explicit is better than implicit in this case, especially for new developer trying to add another new test.